src/python/turicreate/data_structures/sframe.py

```diff
@@ -5881,6 +5881,17 @@ def _group(self, key_columns):
         """
         gsf = GroupedSFrame(self, key_columns)
         return gsf
+    
+    def to_dict(self):
+        dict_sframe = {}
+        for key in self.column_names():
+            sa = SArray(self[key])
+            val_list = []
+            for val in sa:
```

Comment:
Hmm; Pandas actually seems to behave a bit differently. I'm not sure why but it gives dictionaries of dictionaries (where the inner dictionaries have numeric keys ranging from 0 to n for n rows), instead of dictionaries of lists. I suspect this is to better handle sparse dataframes? Example of this behavior:

```python
In [23]: d = df.to_dict()

In [24]: type(d)
Out[24]: dict

In [25]: d.keys()
Out[25]: ['path', 'image']

In [26]: type(d['path'])
Out[26]: dict

In [27]: d['path'].keys()
Out[27]: 
[0,
 1,
 2,
...
```

In any case, we should probably preserve the same behavior here -- let's use a dict instead of a list for the inner structure, using row index as the key.

Reply:
I'm not sure I understand pandas' behavior here.

Reply:
Pandas always gives a sparse representation (even if the original data was dense) -- i.e. using dictionaries instead of lists, with row index as the dictionary key.

Reply:
@znation - what do you recommend we do here? Should we modify this pull so we match pandas' behavior? Should we close the issue (#1909) as will not fix?

Reply:
I'm not sure I have an opinion on this one. I am not a fan of Pandas behavior here, but the goal of #1909 was to match Pandas behavior.